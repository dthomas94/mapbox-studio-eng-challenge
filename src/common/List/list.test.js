import React from "react";
import { List } from ".";
import { mount } from "enzyme";
import { mountToJson } from "enzyme-to-json";

const defaultProps = {
	title: "Favorites List",
	items: [],
	onClickListItem: jest.fn(),
	renderComponent: jest.fn(),
};

describe("List", () => {
	let wrapper;
	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = mount(<List title="Favorites List" {...defaultProps} />);
	});
	it("renders", () => {
		expect(mountToJson(wrapper)).toMatchSnapshot();
	});
	it("check the type, value of items and title", () => {
		wrapper = mount(<List {...defaultProps} />);
		expect(wrapper.prop("items")).toEqual(expect.arrayContaining([]));
		expect(wrapper.prop("title")).toEqual(
			expect.stringContaining("Favorites List")
		);
	});
	it("check the onClickListItem callback", () => {
		const props = {
			items: [{ name: "Default", type: "Default" }],
		};
		wrapper = mount(<List {...defaultProps} items={props.items} />);
		const ListItem = wrapper.find("li");
		ListItem.simulate("click", {
			item: props.items[0],
		});
		expect(defaultProps.onClickListItem).toHaveBeenCalledWith({
			name: "Default",
			type: "Default",
		});
	});
});
