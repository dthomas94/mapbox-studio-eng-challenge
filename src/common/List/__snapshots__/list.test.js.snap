// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders 1`] = `
<List
  items={Array []}
  onClickListItem={[MockFunction]}
  renderComponent={[MockFunction]}
  title="Favorites List"
>
  <div
    className="list"
  >
    <h3>
      Favorites List
    </h3>
    <ul />
  </div>
</List>
`;
